src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -0,0 +1,511 @@
+    }
+
+    @Test
+    @DisplayName("Test indexOf() null element")
+    public void testIndexOfNullElement() {
+        studentsList.add(1, null);
+        assertEquals(studentsList.indexOf(null), 1);
+    }
+
+    @Test
+    @DisplayName("Test lastIndexOf()")
+    public void testLastIndexOf() {
+        assertEquals(studentsList.lastIndexOf(student2), studentsList.size() - studentsList.size() + 1);
+    }
+
+    @Test
+    @DisplayName("Test lastIndexOf null element")
+    public void testLastIndexOfNull() {
+        emptyList.add(null);
+        assertEquals(0, emptyList.lastIndexOf(null));
+    }
+
+    @Test
+    @DisplayName("test returning -1 if provided non existing object to lastIndexOf()")
+    public void testReturning1IfProvidedNonExistingObjectToLastIndexOf() {
+        assertEquals(-1, studentsList.lastIndexOf(student0));
+    }
+
+    @Test
+    @DisplayName("Contains All test")
+    public void containsAllTest() {
+        assertTrue(studentsList.containsAll(listOfStudents()));
+    }
+
+    @Test
+    @DisplayName("Contains All when not all present")
+    public void containsNotAllTest() {
+        assertFalse(studentsList.containsAll(Arrays.asList(student1, student2, student0)));
+    }
+
+    @Test
+    @DisplayName("Check addAll to index")
+    public void checkAddAllToIndex() {
+        emptyList = new StudentList();
+        emptyList.add(student0);
+        emptyList.add(new Student("Karl", LocalDate.parse("2020-01-05"), "Motivated"));
+        emptyList.add(new Student("Colombo", LocalDate.parse("1940-01-05"), "Investigator"));
+        emptyList.addAll(1, listOfStudents());
+
+        assertTrue(emptyList.containsAll(listOfStudents()));
+    }
+
+    @Test
+    @DisplayName("Check deletion the collection of objects")
+    public void checkDeletionTheCollectionOfObjects() {
+        studentsList.add(new Student("Karl", LocalDate.parse("2020-01-05"), "Motivated"));
+        studentsList.add(new Student("Colombo", LocalDate.parse("1940-01-05"), "Investigator"));
+
+        assertTrue(studentsList.removeAll(listOfStudents()));
+    }
+
+    @Test
+    @DisplayName("Test retaining the objects")
+    public void testRetainingTheObjects() {
+        studentsList.add(new Student("Karl", LocalDate.parse("2020-01-05"), "Motivated"));
+        studentsList.add(new Student("Colombo", LocalDate.parse("1940-01-05"), "Investigator"));
+        List<Student> sublistOfStudents = studentsList.subList(2, 5);
+
+        assertTrue(studentsList.retainAll(sublistOfStudents));
+        assertEquals(studentsList.size(), 3);
+        assertEquals(sublistOfStudents.get(0), studentsList.get(0));
+        assertEquals(sublistOfStudents.get(1), studentsList.get(1));
+        assertEquals(sublistOfStudents.get(2), studentsList.get(2));
+    }
+
+    @Test
+    @DisplayName("Check hasNext() of listIterator() when collection is not empty")
+    public void checkHasNextListIterator() {
+        assertFalse(!studentsList.listIterator().hasNext());
```

Comment:
please go through all the code and fix code smells, they are highlighted by IDE
![image](https://user-images.githubusercontent.com/65505271/135151609-d2645c21-b348-4203-b3c4-4cb09d8efabf.png)

Reply:
I have already done it.
The interesting thing is that in most cases Idea suggests changing the simplification back as it was before the first simplification.